features/step_definitions/download_endpoint_steps.rb

```diff
@@ -0,0 +1,28 @@
+require 'open-uri'
+require 'tempfile'
+
+Given(/^I try download "([^"]*)" from channel "([^"]*)"$/) do |rpm, channel|
+  url = "#{Capybara.app_host}/rhn/manager/download/#{channel}/getPackage/#{rpm}"
+  if @token
+    url = "#{url}?#{@token}"
+  end
+  puts url
+  Tempfile.open(rpm) do |tmpfile|
+    @download_path = tmpfile.path
+    begin
+      open(url) do |urlfile|
+        tmpfile.write(urlfile.read)
```

Comment:
I am not sure, but shouldn't you call `tmpfile.unlink` after its [finalization](http://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html)?